packages/neon_framework/lib/src/storage/request_cache.dart

```diff
@@ -63,30 +65,32 @@ final class DefaultRequestCache implements RequestCache {
     final cacheDir = await getApplicationCacheDirectory();
     database = await openDatabase(
       p.join(cacheDir.path, 'cache.db'),
-      version: 2,
+      version: 3,
       onCreate: onCreate,
       onUpgrade: (db, oldVersion, newVersion) async {
-        final batch = db.batch();
-        if (oldVersion == 1) {
-          batch
-            ..execute('ALTER TABLE cache ADD COLUMN etag TEXT')
-            ..execute('ALTER TABLE cache ADD COLUMN expires INTEGER');
-        }
-        await batch.commit();
+        await db.transaction((txn) async {
+          // We can safely drop the table as it only contains cached data.
+          // Non breaking migrations should not drop the cache. The next
+          // breaking change should remove all non breaking migrations before it.
```

Comment:
The comment is better placed before the transaction to explain what the whole onUpgrade method does.